sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -715,49 +667,136 @@ public void validate(PBegin input) {
         }
       }
 
+      private BigQuerySourceBase applySourceTransform(
+          Pipeline p, String jobUuid, String extractDestinationDir) {
+        BigQuerySourceBase source;
+        BigQueryOptions bqOptions = p.getOptions().as(BigQueryOptions.class);
+        final BigQueryServices bqServices = getBigQueryServices();
+        final String executingProject = bqOptions.getProject();
```

Comment:
I'm generally leery of using the options that are available at application time, mainly because I'm trying to get rid of the ability to access them. The graph in the runner API doesn't have options associated with it, so if it's not being explicitly passed it probably shouldn't be used in the `expand()`. Can the `QuerySource` construct its own temp ref in `splitIntoBundles` or `createReader`?